tests/framework/extensions/workloads/pods/verify.go

```diff
@@ -0,0 +1,68 @@
+package pods
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	provv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	projectv3 "github.com/rancher/rancher/pkg/client/generated/project/v3"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	steveV1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	DaemonsetSteveType = "apps.daemonset"
+)
+
+// VerifyReadyDaemonsetPods tries to poll the Steve API to verify the expected number of daemonset pods are in the Ready
+// state
+func VerifyReadyDaemonsetPods(t *testing.T, client *rancher.Client, cluster *v1.SteveAPIObject) error {
+	status := &provv1.ClusterStatus{}
+	err := steveV1.ConvertToK8sType(cluster.Status, status)
+	require.NoError(t, err)
+
+	daemonsetequals := false
+
+	err = wait.Poll(500*time.Millisecond, 5*time.Minute, func() (dameonsetequals bool, err error) {
+		daemonsets, err := client.Steve.SteveType(DaemonsetSteveType).ByID(status.ClusterName)
+		if err != nil {
+			return false, nil
+		}
+
+		daemonsetsStatusType := &projectv3.DaemonSetStatus{}
+		err = v1.ConvertToK8sType(daemonsets.Status, daemonsetsStatusType)
+		if err != nil {
+			return false, nil
+		}
+		if daemonsetsStatusType.DesiredNumberScheduled == daemonsetsStatusType.NumberAvailable {
+			return true, nil
+		}
+		return false, err
+	})
+	if err != nil {
+		return fmt.Errorf("error waiting for daemonset pods to be ready: %v", err)
+	}
+
+	daemonsets, err := client.Steve.SteveType(DaemonsetSteveType).ByID(status.ClusterName)
+	if err != nil {
+		return err
+	}
+
+	daemonsetsStatusType := &projectv3.DaemonSetStatus{}
+	err = v1.ConvertToK8sType(daemonsets.Status, daemonsetsStatusType)
+	if err != nil {
+		return err
+	}
+
+	if daemonsetsStatusType.DesiredNumberScheduled == daemonsetsStatusType.NumberAvailable {
+		daemonsetequals = true
+	}
+
+	assert.Truef(t, daemonsetequals, "Ready Daemonset Pods didn't match expected")
+	return err
```

Comment:
is there some flakiness to the poll function above this that runs the same code? instead of asserting true, we could just assert there's no error after the poll, right?

Reply:
I'm pretty sure in an earlier review it was requested to change it to this as you see it because of flakiness